src/main/java/bio/terra/stairway/HookWrapper.java

```diff
@@ -0,0 +1,70 @@
+package bio.terra.stairway;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HookWrapper {
+  private static final Logger logger = LoggerFactory.getLogger(HookWrapper.class);
+  private Stairway stairway;
+
+  public HookWrapper(Stairway stairway) {
+    this.stairway = stairway;
```

Comment:
Should we create a copy or make stairway read-only? I don't think we want to modify the original stairway object accidentally.

Reply:
With the other changes, I am only passing through the StairwayHooks object, so this shouldn't be an issue anymore!